core/src/main/java/org/apache/iceberg/catalog/BaseSessionCatalog.java

```diff
@@ -84,6 +86,11 @@ public TableBuilder buildTable(TableIdentifier ident, Schema schema) {
       return BaseSessionCatalog.this.buildTable(context, ident, schema);
     }
 
+    @Override
+    public void initialize(String catalogName, Map<String, String> props) {
+      BaseSessionCatalog.this.initialize(catalogName, props);
```

Comment:
This would re-initialize the wrapped `SessionCatalog` so I don't think that this is correct. Instead, this should throw an exception that complains about attempting to initialize the catalog when the underlying session catalog has been initialized.

When used through `RESTCatalog`, this is never called.